src/front/ccrlExtensions.js

```diff
@@ -8,240 +8,397 @@
+    const units = ['k', 'M', 'B', 'T'];
+    let unitIndex = -1;
+    while (number >= 1000 && unitIndex < units.length - 1) {
+      number /= 1000;
+      unitIndex++;
+    }
+    return `${number.toFixed(2)}${units[unitIndex]}`;
+  };
+
+  const getFullmoveNumberFromFen = (fen) => {
+    const parts = fen.split(" ");
+    return parts.length === 6 ? parseInt(parts[5]) : 1;
+  };
+
+  const convertPVToAlgebraic = (pv, startingFen) => {
+    if (typeof Chess === 'undefined') {
+      console.error("Chess.js not loaded");
+      return pv;
+    }
+    const chess = new Chess(startingFen);
+    const moves = pv.trim().split(/\s+/);
+    let groups = [];
+    let currentMoveNumber = getFullmoveNumberFromFen(startingFen);
+    let isWhiteToMove = chess.turn() === 'w';
+
+    moves.forEach((move, i) => {
+      let moveObj = chess.move(move, { sloppy: true });
+      if (!moveObj) return;
+
+      if (isWhiteToMove) {
+        groups.push({
+          moveNumber: currentMoveNumber,
+          white: moveObj.san,
+          black: null
+        });
+      } else {
+        if (groups.length > 0) {
+          groups[groups.length - 1].black = moveObj.san;
+        } else {
+          groups.push({
+            moveNumber: currentMoveNumber,
+            white: "...",  // <-- Place "..." after move number when Black moves first
+            black: moveObj.san
+          });
+        }
+        currentMoveNumber++;
+      }
+
+      isWhiteToMove = chess.turn() === 'w';
+    });
+
+    return groups.map(group => {
+      let moveNumberHtml = `<span style="color: #f5c276; font-weight: bold;">${group.moveNumber}.</span>`;
+      let whiteMove = group.white ? `<span style="color: #ffffff;">${group.white}</span>` : '';
+      let blackMove = group.black ? `<span style="color: #ffffff;">${group.black}</span>` : '';
+
+      return `${moveNumberHtml} ${whiteMove} ${blackMove}`.trim();
+    }).join(" ");
+  };
+
+  // A helper function to parse the score string into a numeric value and display string
+  const parseScore = (scoreText) => {
+    scoreText = scoreText.trim();
+    console.log(scoreText)
+    // Format: "+MX" where X is a positive integer (mate for white)
+    if (scoreText.startsWith("+M")) {
+      const mateMoves = parseInt(scoreText.substring(2));
+
+      // Use a high constant (e.g. 1000) so that mate scores are always above normal evals.
+      const value = 50 - (isNaN(mateMoves) ? 0 : mateMoves * 0.01);
+      return { value: value, tooltip: scoreText };
+    }
+    // Format: "-MX" where X is a positive integer (mate for black)
+    else if (scoreText.startsWith("-M")) {
+      const mateMoves = parseInt(scoreText.substring(2));
+      // Use a low constant so that mate scores are always below normal evals.
+      const value = -50 + (isNaN(mateMoves) ? 0 : mateMoves * 0.01);
+      return { value: value, tooltip: scoreText };
+    }
+    else {
+      const numStr = scoreText;
+      const value = parseFloat(numStr).clamp(-20,20);
```

Comment:
As discussed, this fails for me but not for Gio.
Let's do `Math.min(Math.max(parseFloat(numStr), -20), 20);` to play it safe, I guess, unless we can think of a more clever idea/solution.

Reply:
Solved by 110b0bf8400f5aa5e2cb8c5c583c8d3ddef6b139